src/tests/adata.c

```diff
@@ -216,11 +217,16 @@ display_authdata(krb5_authdata *ad, krb5_keyblock *skey, krb5_keyblock *tktkey,
         ad->ad_type == KRB5_AUTHDATA_CAMMAC) {
         /* Decode and display the contents. */
         inner_ad = get_container_contents(ad, skey, tktkey);
-        display_authdata_list(inner_ad, skey, tktkey, get_prefix_byte(ad));
+        display_authdata_list(inner_ad, skey, tktkey, get_prefix_byte(ad),
+                              (ad->ad_type == KRB5_AUTHDATA_IF_RELEVANT &&
```

Comment:
I think it'd be clearer to give this condition its own variable.

Reply:
I've updated the tests slightly.